templates/content-certificate.php

```diff
@@ -38,19 +42,29 @@
 	</div>
 </div>
 
-<div class="llms-print-certificate no-print" id="llms-print-certificate">
-	<button class="llms-button-secondary" onClick="window.print()" type="button">
-		<?php echo _e( 'Print', 'lifterlms' ); ?>
-		<i class="fa fa-print" aria-hidden="true"></i>
-	</button>
-
-	<form action="" method="POST">
-		<button class="llms-button-secondary" type="submit" name="llms_generate_cert">
-			<?php echo _e( 'Save', 'lifterlms' ); ?>
-			<i class="fa fa-cloud-download" aria-hidden="true"></i>
+<?php
+if ( $is_owner_or_admin ) {
+?>
+	<div class="llms-print-certificate no-print" id="llms-print-certificate">
+		<button class="llms-button-secondary" onClick="window.print()" type="button">
+			<?php echo _e( 'Print', 'lifterlms' ); ?>
+			<i class="fa fa-print" aria-hidden="true"></i>
 		</button>
 
-		<input type="hidden" name="certificate_id" value="<?php echo get_the_ID(); ?>">
-		<?php wp_nonce_field( 'llms-cert-actions', '_llms_cert_actions_nonce' ); ?>
-	</form>
-</div>
+		<form action="" method="POST">
+			<button class="llms-button-secondary" type="submit" name="llms_generate_cert">
+		  	<?php echo _e( 'Save', 'lifterlms' ); ?>
+				<i class="fa fa-cloud-download" aria-hidden="true"></i>
+			</button>
+
+			<?php echo _e( 'Make this link public', 'lifterlms' ); ?>
+			<input type="checkbox" name="llms_is_shareable_cert" <?php echo ( $is_shareable ? 'checked' : '' ); ?> onchange="document.getElementsByName('llms_change_shareable_cert')[0].click()" />
+			<button class="llms-button-secondary" style="visibility: hidden;" type="submit" name="llms_change_shareable_cert"></button>
```

Comment:
Something about using inline JS and a hidden button with no content feels "hacky" (for lack of a better word).

I think that this would *feel* better to me if there was a button that acted as a toggle depending on the current state of the certificate's sharing settings.

Something like:

```php
<button ...><?php echo llms_parse_bool( $cert->get( 'allow_sharing' ) ? 'Enable sharing' : 'Disable Sharing'; ?></button>
```

What do you think?

Reply:
it's ok for me, but i thought the checkbox will be more obvious in favor of change the text of the button

and in button way i have to make tow different actions one for enable and one for disable the sharing settings